js/src/hooks/useAutoCheckAdsAccountStatus.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+import { __ } from '@wordpress/i18n';
+import { useCallback, useRef } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { useAppDispatch } from '.~/data';
+import { API_NAMESPACE } from '.~/data/constants';
+import useDispatchCoreNotices from '.~/hooks/useDispatchCoreNotices';
+import useWindowFocusCallbackIntervalEffect from '.~/hooks/useWindowFocusCallbackIntervalEffect';
+
+const useAutoCheckAdsAccountStatus = ( onAccessGranted = noop ) => {
+	const { createNotice } = useDispatchCoreNotices();
+	const { receiveAdsAccountStatus } = useAppDispatch();
+	const hasAccessRef = useRef();
+
+	const onAccessGrantedRef = useRef();
+	onAccessGrantedRef.current = onAccessGranted;
+
+	const checkStatus = useCallback( async () => {
+		const prevHasAccess = hasAccessRef.current;
+		const accountStatus = await apiFetch( {
+			path: `${ API_NAMESPACE }/ads/account-status`,
+		} );
```

Comment:
Unfortunately no @joemcgill , because the datastore is not up to date and we'll have to invalidate the data each time. The pattern is the same one used elsewhere in the codebase. For e.g https://github.com/woocommerce/google-listings-and-ads/blob/a6e8a4c2be6da1a5a2b73602991d3384db74f8dc/js/src/components/paid-ads/billing-card/useAutoCheckBillingStatusEffect.js#L27